### ISSUE TITLE
Getting the Maximum Number of Iterations Needed to Zoom to a Point

### ISSUE THREAD

**Issue by ShiromMakkad** [adelelopez/antelbrot#3]
I've been looking for a clean and easy way to get the maximum number of iterations, but everything out there is extremely complex, except yours. 

The line:
if (re > 1024 || im > 1024 || re < -1024 || im < -1024)
            return v;

allows you to easily find the truncate maximum number of iterations needed for a point, and it works better than programs like Ultra Fractal do it. 

How did you come up with this and more importantly what is the significance of 1024? I want to know whether this can be universally applied or just up to a certain point. So far, it's worked on every depth I've tried. Any help would be appreciated. Thanks!

**Comment by ethan-dorta** [comment]
I know 1024 is a power of 2, which could optimize it for bitwise operations.

**Comment by adelelopez** [comment]
I think I determined it through trial and error. 

If your deep zoom point is within the set (which you want it to be), then the iterations of the point should never have absolute value greater than 2, and so `re` and `im` should never be above 4.

But when you're trying to click on a point to zoom in on (which gets used as the next deep zoom point), it can sometimes be hard to make sure it's actually in the set. So expanding the check to 1024 just gives us some leeway. The main thing is just that there *is* some bound check, otherwise it will go to infinity which will most likely crash the program.

**Comment by ShiromMakkad** [comment]
When you run the code, does your maximum iteration count scale with the magnification?

**Comment by adelelopez** [comment]
No, it's limited to 1000 (https://github.com/adelelopez/antelbrot/blob/master/antelbrot.cpp#L164) unless you input a new one.

**Comment by ShiromMakkad** [comment]
That's very odd. I guess it isn't automatically determining the max iteration in the same way then for you. I've added a piece in the code that manually sets the precision of the numbers based on the magnification. I think that's what's causing this "feature." I'm going to investigate further into this, because this could be really helpful for a lot of fractal renderers. I'll update this post once I've figured it out.
